backend/tests/auth_api.test.ts

```diff
@@ -0,0 +1,271 @@
+import { test, afterEach, after, describe } from "node:test";
+import assert from "node:assert";
+import supertest from "supertest";
+import mongoose from "mongoose";
+import User from "../models/user";
+import app from "../app";
+import * as helper from "./test_helper";
+
+const api = supertest(app);
+
+describe("POST /auth/login", () => {
+	afterEach(async () => {
+		await User.deleteMany({});
+	});
+	test("retuns correct data when the credentials are correct", async () => {
+		await helper.createTestUser();
+
+		const response = await helper.postLogin(api, {
+			username: helper.TEST_USER.valid.username,
+			password: helper.TEST_USER.valid.password,
+		});
+
+		const { accessToken, refreshToken, username, name } = response.body;
+
+		assert.strictEqual(username, helper.TEST_USER.valid.username.toLowerCase());
+		assert.strictEqual(name, helper.TEST_USER.valid.name);
+		assert.ok(accessToken);
+		assert.ok(
+			refreshToken && refreshToken !== helper.TEST_USER.valid.refreshToken
+		);
+
+		const accessTokenUsername = helper.getUsernameFromToken(accessToken);
+		assert.strictEqual(
+			accessTokenUsername,
+			helper.TEST_USER.valid.username.toLowerCase()
+		);
+
+		const refreshTokenUsername = helper.getUsernameFromToken(refreshToken);
+		assert.strictEqual(
+			refreshTokenUsername,
+			helper.TEST_USER.valid.username.toLowerCase()
+		);
+	});
+
+	test("ignores cases in username and returns correct data with username in lowercase", async () => {
+		await helper.createTestUser();
+
+		const response = await helper.postLogin(api, {
+			username: helper.TEST_USER.valid.upperCaseUsername,
+			password: helper.TEST_USER.valid.password,
+		});
+
+		const { username } = response.body;
+
+		assert.strictEqual(username, helper.TEST_USER.valid.username.toLowerCase());
+	});
+
+	test("returns 401 code and error message when the username or password is wrong", async () => {
+		const invalidUsernameResponse = await helper.postLogin(
+			api,
+			{
+				username: helper.TEST_USER.invalid.wrongUsername,
+				password: helper.TEST_USER.valid.password,
+			},
+			401
```

Comment:
Probably you need to create a http status code TS enum to replace these, and also avoid using magic number.